### DISCUSSION TITLE
DAI 2023-2024 - Practical work 1 - Colin Jaques

### DISCUSSION THREAD

**Discussion by CoJaques** [heig-vd-dai-course/heig-vd-dai-course#84] [Show and tell]
For the Practical Work 1, I plan to develop a versatile File Encoder/Decoder CLI tool using picocli. The CLI will be designed to take an input and an output file path as its primary arguments. One of the key features I aim to implement is an extensible architecture that allows for easy integration of various encoding and decoding algorithms, starting with Base64 and ROT13.

My focus will be on creating a universally supportive tool that reads and writes in bytes, ensuring it can handle not just text files but also binary files for maximum utility. This will keep the tool performance-optimized while making it extremely adaptable for future use cases.

By keeping modularity and scalability at the forefront of the design, I aim to make this tool a valuable and easily extensible CLI toolkit.

**Comment by ludelafo** [comment]
Seems great! Looking forward to your solution!

**Comment by CoJaques** [comment]
You'll find my work here : https://github.com/CoJaques/Henk

**Reply by CoJaques** [reply]
And my release here -> https://github.com/CoJaques/Henk/releases/tag/Release-V1.0.0
